internal/test/wmcb/wmcb_test.go

```diff
@@ -677,3 +677,42 @@ func (f *wmcbFramework) getLatestCniPluginsSHA() (string, error) {
 	sha512 := sha512AndFilename[0]
 	return sha512, nil
 }
+
+// initializeKubeNodePkgInfo populates the fileds of KubeNode pkg
+func (f *wmcbFramework) initializeKubeNodePkgInfo() error {
+	f.kubeNode = pkgInfo{}
+
+	f.kubeNode.url = kubeNodeBaseURL + framework.K8sVersion + "/kubernetes-node-windows-amd64.tar.gz"
+
+	kubeNodeSHA, err := f.getKubeNodeSHA()
+	if err != nil {
+		return fmt.Errorf("unable to get SHA for Kubernetes node binary for Windows: %v", err)
+	}
+	f.kubeNode.sha = kubeNodeSHA
+
+	f.kubeNode.shaType = "sha512"
+	return nil
+}
+
+// getKubeNodeSHA returns the SHA512 value of the kube node package.
+func (f *wmcbFramework) getKubeNodeSHA() (string, error) {
+
+	kubeNodeChecksumFileURL := f.kubeNode.url + ".sha512"
+	// Fetching checksum file
+	response, err := http.Get(kubeNodeChecksumFileURL)
+	if err != nil {
+		return "", fmt.Errorf("could not fetch kube node binary checksum file: %v", err)
+	}
+	defer response.Body.Close()
+	// the checksum file content contains a single <sha> value
```

Comment:
This comment would be better placed before `for scanner.Scan()` as that is where this assumption takes effect. You should also mention that you expect an EOF in the first line itself.

Reply:
Thinking a bit more on this, it would be safer for you to have a guard against the contents being in an unexpected format. Otherwise in the scenario when the checksum file contents have changed, the test would fail in an unexpected manner.

Reply:
It's also better to write an example of how contents would look like in the comments, so that while reviewing we can see if have more validations.